pkg/broker/handler/handler_test.go

```diff
@@ -141,6 +138,91 @@ func TestHandler(t *testing.T) {
 	})
 }
 
+type alwaysErrProc struct {
+	processors.BaseProcessor
+	desiredErrCount, currErrCount int
+}
+
+func (p *alwaysErrProc) Process(_ context.Context, _ *event.Event) error {
+	if p.currErrCount < p.desiredErrCount {
+		p.currErrCount++
+		return errors.New("always error")
+	}
+	return nil
```

Comment:
The processor can store the event in a channel once succeeding and the test can wait for the event.